azure_functions/function_app.py

```diff
@@ -144,6 +187,37 @@ def alert_to_slack(req: func.HttpRequest) -> func.HttpResponse:
         return func.HttpResponse(f"Error sending to Slack: {e}", status_code=500)
 
 
+@app.route(route="alert_to_slack", auth_level=func.AuthLevel.ANONYMOUS, methods=["POST"])
+def alert_to_slack(req: func.HttpRequest) -> func.HttpResponse:
+    """
+    Receives an alert from Azure Monitor, formats it, and sends it to Slack via webhook.
+    """
+    logging.info("alert_to_slack received a request.")
+
+    provided_code = req.params.get("code")
+    auth_response = validate_function_key(provided_code)
+    if auth_response:
+        return auth_response
+
+    try:
+        alert_payload = req.get_json()
+    except ValueError:
+        return func.HttpResponse("Request body is not valid JSON.", status_code=400)
+
+    data = alert_payload.get("data", {})
+
+    data_str = json.dumps(data, indent=2)  # pretty print 2 spaces indent
+    if len(data_str) > 10000:  # 1 byte per char, roughly 10KB, Azure limits to 64KB
+        data_str = data_str[:10000] + "... [truncated]"
+    logging.info(f"Received Azure alert data:\n{data_str}")

```

Comment:
I know you had some logging in here for debugging etc. Do we still need this? If it is still useful, maybe we just log as a `debug` message instead, so we can filter it out later and/or turn up the logging threshold to avoid these in e.g. production?

Reply:
That makes sense! I changed this (and one other log message) to `debug` in 66563b8757391e0e3f1755bd0c186c818b5ed0cd in case we need to turn them on again later. For example if we ever see a new type of Application Insights alert come in and want to check it in the logs.